src/Slices/Dashboard/UI/helper.ts

```diff
@@ -61,12 +61,79 @@ export const formatMetricsToStacked = (
           max = tempMax;
         }
       });
+      tempCharState = tempCharState.map((metric) => formatValues(metric));
     }
   } else {
-    tempCharState = [metrics as Metric];
+    tempCharState = [
+      metrics.name.includes("service_count")
+        ? formatValues(metrics as Metric)
+        : (metrics as Metric),
+    ];
     max = (metrics as Metric).data
       .flatMap((value) => (value !== null ? value : 0))
       .sort((a, b) => a - b)[(metrics as Metric).data.flat().length - 1];
   }
   return [tempCharState, max];
 };
+
+const formatValues = (metrics: Metric) => {
+  const newMetrics = metrics.data.map((data, index) => {
+    if (data == null || index === 0) {
+      return null;
+    }
+    if (metrics.data[index - 1] !== null) {
+      if (data > (metrics.data[index - 1] as number)) {
+        return Math.floor(data);
+      } else {
+        return Math.ceil(data);
+      }
+    }
+    return Math.round(data);
+  });
+  return {
+    ...metrics,
+    data: newMetrics,
+  };
+};
+
+/**
+ * Replace null values between two numbers(no matter how many nulls they are in between,
+ * as long as there are some boundaries) with interpolation on line between those two numbers.
+ *
+ * @param metrics
+ * @returns metrics with interpolated values instead null values
+ */
+export const interpolateMetrics = (metrics: (number | null)[]) => {
+  const newMetric: (number | null)[] = [];
+  let nextNumber = 0;
+
+  metrics.forEach((value, index) => {
```

Comment:
I think this piece of code can use a few comments to explain the logic and reasoning :)

Reply:
Yup, that's right, I didn't think of that :D